src/System.Linq.Parallel/src/System/Linq/Parallel/Channels/AsynchronousChannel.cs

```diff
@@ -435,7 +441,7 @@ internal bool TryDequeue(ref T item)
         //     True if a chunk was found, false otherwise.
         //
 
-        private bool TryDequeueChunk(ref T[] chunk)
+        private bool TryDequeueChunk(ref T[]? chunk)
```

Comment:
💭 Should this have `NotNullWhen(true)`?

Reply:
No because InternalDequeueChunk() method return nullable:
```csharp
chunk = InternalDequeueChunk();
return true```

Reply:
📝 This comment is paired with https://github.com/dotnet/corefx/pull/41533#discussion_r332177769, which questions the need for the nullable return type on `InternalDequeueChunk`.

Reply:
I think your comment #41533 is the right assumption and this method should be updated accordingly